tests/scene/test_code_edit.h

```diff
@@ -2691,6 +2691,132 @@ TEST_CASE("[SceneTree][CodeEdit] folding") {
 	memdelete(code_edit);
 }
 
+TEST_CASE("[SceneTree][CodeEdit] region folding") {
+	CodeEdit *code_edit = memnew(CodeEdit);
+	SceneTree::get_singleton()->get_root()->add_child(code_edit);
+	code_edit->grab_focus();
+
+	SUBCASE("[CodeEdit] region folding") {
```

Comment:
Some addition tests:

- create_code_region
-- No selection
-- No valid comment delimiter
-- mutli-carets

Case when we have an existing code region then the tag changes.